configure

```diff
@@ -0,0 +1,5018 @@
+#! /bin/sh
+# Guess values for system-dependent variables and create Makefiles.
+# Generated by GNU Autoconf 2.69 for py-libzfs 1.0.
+#
+#
+# Copyright (C) 1992-1996, 1998-2012 Free Software Foundation, Inc.
+#
+#
+# This configure script is free software; the Free Software Foundation
+# gives unlimited permission to copy, distribute and modify it.
+## -------------------- ##
+## M4sh Initialization. ##
+## -------------------- ##
+
+# Be more Bourne compatible
+DUALCASE=1; export DUALCASE # for MKS sh
+if test -n "${ZSH_VERSION+set}" && (emulate sh) >/dev/null 2>&1; then :
+  emulate sh
+  NULLCMD=:
+  # Pre-4.2 versions of Zsh do word splitting on ${1+"$@"}, which
+  # is contrary to our usage.  Disable this feature.
+  alias -g '${1+"$@"}'='"$@"'
+  setopt NO_GLOB_SUBST
+else
+  case `(set -o) 2>/dev/null` in #(
+  *posix*) :
+    set -o posix ;; #(
+  *) :
+     ;;
+esac
+fi
+
+
+as_nl='
+'
+export as_nl
+# Printing a long string crashes Solaris 7 /usr/bin/printf.
+as_echo='\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\'
+as_echo=$as_echo$as_echo$as_echo$as_echo$as_echo
+as_echo=$as_echo$as_echo$as_echo$as_echo$as_echo$as_echo
```

Comment:
Can you please explain what this is good for? Why exactly this many backslashes?

Reply:
Configure file is dynamically generated by autoconf - this is probably being done to support cross platform runs ?

Reply:
I agree with @igalic that this file should be excluded from the git repository.

Reply:
There is no harm in having it here.

Reply:
The file is too complex to reasonably review it. The work overhead of keeping it audited is pretty  high, so that there is the risk of malicious code within the file. For security reasons I recommend to drop the file from the source repository.

Reply:
Thank you for your opinion but we are going to keep it. Security is not a concern for us.
If cpython and dozens of other projects do it, I dont see why we should do differently for these reasons.

Reply:
> Security is not a concern for us.

For some of us it is. At the current time [libiocage](https://github.com/iocage/libiocage/) relies on this module with elevated privileges. I would prefer py-libzfs to contain no code that the authors did not understand. And in case of a auto-generated file, I don't see the reason to raise the overhead of understanding each and every bit of it versus letting it become generated by autoconf from files that are more human-readable.

> If cpython and dozens of other projects do it, I dont see why we should do differently for these pointed out reasons.
IMO they are doing it wrong or have pretty aware reviewers.

Reply:
You are not required to use it, you can generate it yourself if you dont want to audit configure.

Reply:
python's configure.ac is probably the cleanest m4 code i've ever seen in my life, and still, it's configure script is 18k lines. but, what they somehow manage is this:

```
 ~/s/p/cpython  ⎇  rm configure
 ~/s/p/cpython  ⎇ *  autoreconf -if
 ~/s/p/cpython  ⎇ 
 ~/s/p/cpython  ⎇  git diff
 ~/s/p/cpython  ⎇  # nothing to see here
```

Reply:
> You are not required to use it, you can generate it yourself if you dont want to audit configure.

That works for me! I'd love to see it being used by the ports package:

```
USES=		python
```
can be changed to
```
USES=		autoreconf python
```